pkg/cloud/aws/sqs/service.go

```diff
@@ -0,0 +1,282 @@
+		},
+	})
+
+	return props, err
+}
+
+func (s *service) QueueExists(ctx context.Context, name string) (bool, error) {
+	s.logger.WithFields(log.Fields{
+		"name": name,
+	}).Info("checking the existence of sqs queue")
+
+	var err error
+	var url string
+
+	if url, err = s.GetUrl(ctx, name); err != nil {
+		return false, fmt.Errorf("can not get url of queue: %w", err)
+	}
+
+	if len(url) > 0 {
+		s.logger.Info("found queue %s with url %s", name, url)
+		return true, nil
+	}
+
+	s.logger.WithFields(log.Fields{
+		"name": name,
+	}).Info("could not find queue")
+
+	return false, nil
+}
+
+func (s *service) GetPropertiesByArn(ctx context.Context, arn string) (*Properties, error) {
+	i := strings.LastIndex(arn, ":")
+	name := arn[i+1:]
+
+	var err error
+	var url string
+
+	if url, err = s.GetUrl(ctx, name); err != nil {
+		return nil, fmt.Errorf("can not get url: %w", err)
+	}
+
+	return &Properties{
+		Name: name,
+		Url:  url,
+		Arn:  arn,
+	}, nil
+}
+
+func (s *service) GetPropertiesByName(ctx context.Context, name string) (*Properties, error) {
+	url, err := s.GetUrl(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	arn, err := s.GetArn(ctx, url)
+	if err != nil {
+		return nil, err
+	}
+
+	properties := &Properties{
+		Name: name,
+		Url:  url,
+		Arn:  arn,
+	}
+
+	return properties, nil
+}
+
+func (s *service) GetUrl(ctx context.Context, name string) (string, error) {
+	var err error
+	var out *sqs.GetQueueUrlOutput
+
+	input := &sqs.GetQueueUrlInput{
+		QueueName: aws.String(name),
+	}
+
+	if out, err = s.client.GetQueueUrl(ctx, input); err != nil {
+		var errQueueDoesNotExist *types.QueueDoesNotExist
+		if errors.As(err, &errQueueDoesNotExist) {
+			return "", nil
+		}
+
+		return "", nil
+	}
+
+	return *out.QueueUrl, nil
+}
+
+func (s *service) GetArn(ctx context.Context, url string) (string, error) {
+	var err error
+	var out *sqs.GetQueueAttributesOutput
+
+	input := &sqs.GetQueueAttributesInput{
+		AttributeNames: []types.QueueAttributeName{"QueueArn"},
+		QueueUrl:       aws.String(url),
+	}
+
+	if out, err = s.client.GetQueueAttributes(ctx, input); err != nil {
+		return "", fmt.Errorf("can not get queue attributes: %w", err)
+	}
+
+	arn := out.Attributes["QueueArn"]
+
+	return arn, nil
+}
+
+func (s *service) Purge(ctx context.Context, url string) error {
+	_, err := s.client.PurgeQueue(ctx, &sqs.PurgeQueueInput{
+		QueueUrl: aws.String(url),
+	})
+
+	return err
+}
+
+func (s *service) createDeadLetterQueue(ctx context.Context, settings *Settings) (map[string]string, error) {
+	attributes := make(map[string]string)
+
+	if !settings.RedrivePolicy.Enabled {
+		return attributes, nil
+	}
+
+	deadLetterName := fmt.Sprintf("%s-dead", settings.QueueName)
+	deadLetterInput := &sqs.CreateQueueInput{
+		QueueName: aws.String(deadLetterName),
+	}
+
+	props, err := s.doCreateQueue(ctx, deadLetterInput)
+	if err != nil {
+		s.logger.Error("could not get arn of dead letter sqs queue %v: %w", deadLetterName, err)
```

Comment:
This log message seems wrong, also we should not log and return the error, but just wrap it I guess

Reply:
fixed